### PR TITLE
feat: Add `primeDivisors` as an abbreviation for `n.factorization.support`

### PR DESCRIPTION
**Pull request by amellendijk**
Replace `n.factorization.support` and `n.factors.toFinset` with `n.primeDivisors` wherever possible. Also rename lemmas that use `factorization` to mean the Finset of prime divisors. 

see [zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/toFinsetFactors)

---

The biggest drawbacks with making it an abbreviation are that (1) we can't eliminate `factors.toFinset` lower down the import hierarchy, notably in `NumberTheory/Divisors`, and (2) we're missing out on an intuitive `simp` lemma that yields `p.Prime ∧ p ∣ n ∧ n ≠ 0` akin to [Nat.mem_divisors](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/Divisors.html#Nat.mem_divisors). That said, people with much more experience than me have expressed a preference for this abbreviation, and indeed this change is relatively painless (modulo some renames).

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Factorization/Basic.lean [hunk 1]
```diff
@@ -122,27 +122,39 @@ theorem factorization_zero : factorization 0 = 0 := by simp [factorization]
...
 theorem support_factorization {n : ℕ} : n.factorization.support = n.factors.toFinset := by
   simp [factorization]
 #align nat.support_factorization Nat.support_factorization
 
-theorem factor_iff_mem_factorization {n p : ℕ} : p ∈ n.factorization.support ↔ p ∈ n.factors := by
+theorem primeDivisors_eq_factors {n : ℕ} : n.primeDivisors = n.factors.toFinset := rfl
+
+theorem primeDivisors_eq_factorization {n : ℕ} : n.primeDivisors = n.factorization.support := rfl
+
+theorem factor_iff_mem_primeDivisors {n p : ℕ} : p ∈ n.primeDivisors ↔ p ∈ n.factors := by
```
Comment (jcommelin) [inline Mathlib/Data/Nat/Factorization/Basic.lean:143]: This name seems swapped wrt the statement. (I guess it was already wrong before your refactor, but please correct it.)

File: Mathlib/Data/Nat/Factorization/Basic.lean [hunk 2]
```diff
@@ -122,27 +122,46 @@ theorem factorization_zero : factorization 0 = 0 := by simp [factorization]
...
+theorem primeDivisors_eq_factors {n : ℕ} : n.primeDivisors = n.factors.toFinset := rfl
+
+theorem primeDivisors_eq_factorization {n : ℕ} : n.primeDivisors = n.factorization.support := rfl
+
+theorem mem_primeDivisors_iff_factor {n p : ℕ} : p ∈ n.primeDivisors ↔ p ∈ n.factors := by
   simp only [support_factorization, List.mem_toFinset]
-#align nat.factor_iff_mem_factorization Nat.factor_iff_mem_factorization
+#align nat.factor_iff_mem_factorization Nat.mem_primeDivisors_iff_factor
+
+theorem mem_primeDivisors {n p : ℕ} : p ∈ n.primeDivisors ↔ p.Prime ∧ p ∣ n ∧ n ≠ 0 := by
```
[the remark this answers is not in the document]
Comment (erdOne) [reply Mathlib/Data/Nat/Factorization/Basic.lean:147]: Maybe you could instead add a simp lemma as `n.factorization p ≠ 0 ↔ p.Prime ∧ p ∣ n ∧ n ≠ 0`?

[the remark this answers is not in the document]
Comment (erdOne) [reply Mathlib/Data/Nat/Factorization/Basic.lean:147]: Or maybe just don't use `abbrev`?

**Comment by FLDutchmann** [comment]
This issue was solved by #8164

**Pull request closed without merging**